### PR TITLE
Removed default profile functionality

### PR DESCRIPTION
**Pull request by charles-cowart**

### CODE REVIEW

File: sequence_processing_pipeline/Pipeline.py [hunk 1]
```diff
@@ -293,72 +292,33 @@ def _configure_profile(self):
...
-                base_profile = profile
-            else:
-                p_a_type = profile['profile']['assay_type']
+            i_type = profile['profile']['instrument_type']
+            a_type = profile['profile']['assay_type']
 
-                # if both items have been found, it's safe to break early.
-                if base_profile is not None and selected_profile is not None:
-                    break
+            print("PROFILE FOUND: %s, %s" % (i_type, a_type))
```
Comment (antgonza) [inline sequence_processing_pipeline/Pipeline.py:311]: needed?

File: sequence_processing_pipeline/Pipeline.py [hunk 2]
```diff
@@ -293,72 +292,33 @@ def _configure_profile(self):
...
+
+            if i_type == instr_type and a_type == assay_type:
+                selected_profile = profile
+                break
+
+        if selected_profile is None:
+            raise ValueError(f"a matching profile ({instr_type}, {assay_type}"
```
Comment (antgonza) [inline sequence_processing_pipeline/Pipeline.py:318]: maybe add to report it to admin? (AKA us)

File: sequence_processing_pipeline/tests/data/configuration_profiles/miseq_metagenomic.json [hunk 3]
```diff
@@ -14,6 +14,38 @@
...
+        "nprocs": 16,
+        "queue": "qiita",
+        "wallclock_time_in_minutes": 216,
+        "modules_to_load": [
+          "bclconvert_3.7.5"
+        ],
+        "executable_path": "bcl-convert",
+        "per_process_memory_limit": "10gb"
+      },
+      "qc": {
```
Comment (antgonza) [inline sequence_processing_pipeline/tests/data/configuration_profiles/miseq_metagenomic.json:28]: should these section be deleted?

File: sequence_processing_pipeline/tests/data/configuration_profiles/novaseq_amplicon.json [hunk 4]
```diff
@@ -14,6 +14,38 @@
...
+        "nprocs": 16,
+        "queue": "qiita",
+        "wallclock_time_in_minutes": 216,
+        "modules_to_load": [
+          "bclconvert_3.7.5"
+        ],
+        "executable_path": "bcl-convert",
+        "per_process_memory_limit": "10gb"
+      },
+      "qc": {
```
Comment (antgonza) [inline sequence_processing_pipeline/tests/data/configuration_profiles/novaseq_amplicon.json:28]: likewise rm?

**Review by antgonza — CHANGES_REQUESTED** [review]

File: sequence_processing_pipeline/Pipeline.py (hunk 1 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/Pipeline.py:311]: I knew I forgot something.

File: sequence_processing_pipeline/tests/data/configuration_profiles/miseq_metagenomic.json (hunk 3 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/tests/data/configuration_profiles/miseq_metagenomic.json:28]: debated that myself. IMOO, it should be retained as long as the legacy QC implementation is still in the codebase. The question is really whether NuQC is stable enough that we can remove legacy QC from the current codebase. My vote is yes.

File: sequence_processing_pipeline/tests/data/configuration_profiles/novaseq_amplicon.json (hunk 4 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/tests/data/configuration_profiles/novaseq_amplicon.json:28]: see above.

File: sequence_processing_pipeline/tests/data/configuration_profiles/miseq_metagenomic.json (hunk 3 above)
Comment (antgonza) [reply sequence_processing_pipeline/tests/data/configuration_profiles/miseq_metagenomic.json:28]: Agree and vote for a different PR for that.

File: sequence_processing_pipeline/Pipeline.py (hunk 2 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/Pipeline.py:318]: We haven't introduced email notifications into this part of the code, so IMOO we should handle it as a separate issue. I think we would be better served with a chunk of code that catches all (or just many) Errors and sends an email along with displaying the stack trace to the user like we already do.

Comment (antgonza) [reply sequence_processing_pipeline/Pipeline.py:318]: ha! I meant add to the message that they should email the admins and not just ignore

**Review by antgonza — APPROVED** [review]

**Pull request merged**